src/ui/map.js

```diff
@@ -2618,16 +2622,29 @@ class Map extends Camera {
         return this.style.getFeatureState(feature);
     }
 
-    _containerDimensions() {
-        let width = 0;
-        let height = 0;
-
+    _containerDimensions(): any {
         if (this._container) {
-            width = this._container.getBoundingClientRect().width || 400;
-            height = this._container.getBoundingClientRect().height || 300;
+            const width = this._container.getBoundingClientRect().width || 400;
+            const height = this._container.getBoundingClientRect().height || 300;
+
+            let transformValues;
+            let el = this._container;
+            while (el && !transformValues) {
+                const transformMatrix = window.getComputedStyle(el).transform;
+                if (transformMatrix && transformMatrix !== 'none') transformValues = transformMatrix.match(/matrix.*\((.+)\)/)[1].split(', ');
+                el = el.parentElement;
+            }
+
+            if (transformValues) {
+                this._containerWidth = transformValues[0] && transformValues[0] !== '0' ? Math.abs(width / transformValues[0]) : width;
+                this._containerHeight = transformValues[3] && transformValues[3] !== '0' ? Math.abs(height / transformValues[3]) : height;
+            } else {
+                this._containerWidth = width;
+                this._containerHeight = height;
+            }
         }
 
-        return [width, height];
+        return [this._containerWidth, this._containerHeight];
```

Comment:
I think we can change this function to not return anything, since the values are being cached in this already.